### DISCUSSION TITLE
Periodic function in MOOSE.

### DISCUSSION THREAD

**Discussion by AlejandroRiano2023** [idaholab/moose#28268] [Q&A Modules: Phase field] [answered]
Good afternoon, I am trying to implement a periodic function in MOOSE.

![image](https://github.com/user-attachments/assets/50e6247b-ec9e-4489-be0b-8df5235bc496)

Currently, I have defined the red points that can be appreciated in the image, and the idea is to repeat the cycle indefinitely.

This is the function defined in just one cycle for the red points in the image.

      [Delta_def]
        type = PiecewiseConstant
        x = '0 20e6 40e6 60e6 80e6 100e6'
        y = '-0.0047629 -0.0074940 -0.0062045 -0.15252 -0.0090877 -0.0048300'
        direction = right
      []

I will be attentive to your comments and thank you so much in advance for your assessment.

Best regards,
Alejandro

**Comment by GiudGiud** [comment, marked as answer]
Hello

The easiest way to do that will be to use the function you have to define the profile, then use a ParsedFunction with an if condition (or the % operator if it works) to create the periodicity on the initial function

Guillaume
